### ISSUE TITLE
How to show previous coach mark?

### ISSUE THREAD

**Issue by kientux** [ephread/Instructions#113]
I can't find any method to show previous coach mark. I think about a way that is stop the flow then start again and skip some coackmarks.. Is that too complicated? Also I can't find any public property of current coackmark's index. So do I have to use delegate to get it if I want to skip coackmarks using `coachMarksController.flow.showNext(numberOfCoachMarksToSkip:)`?

**Comment by kientux** [comment]
I've just read the source code of method `showNext(numberOfCoachMarksToSkip:)` and found out that it just increase `currentIndex` by `numberToSkip`. So if I comment out the negative check and pass `numberToSkip` = -2, it will show previous coachmark.

The only problem is `showNextCoachMark(hidePrevious: true)` doesn't hide current coachmark.

**Comment by kientux** [comment]
I modified the source code to make it works:

```
// alreadyStarted indicates that the flow is started and won't check for currentIndex = 0
internal func showNextCoachMark(hidePrevious: Bool = true, alreadyStarted: Bool = false) {
    ...
    if currentIndex == 0 && !alreadyStarted {
        createAndShowCoachMark()
        return
    }
    ....
}

open func showNext(numberOfCoachMarksToSkip numberToSkip: Int = 0) {
    if !self.started || !canShowCoachMark { return }

//        if numberToSkip < 0 {
//            print("showNext: The specified number of coach marks to skip" +
//                  "was negative, nothing to do.")
//            return
//        }

    currentIndex += numberToSkip

    showNextCoachMark(hidePrevious: true, alreadyStarted: numberToSkip < 0)
}
```

But it's just a temporary solution and I don't think this is a good way. Maybe a separate method will be better.

**Comment by ephread** [comment]
Hey @kientux, indeed, there are currently no ways to go back. Yes you're on the right track and, yes we'll need a separate method. It should be easy to implement. ;)

**Comment by ephread** [comment]
@robert-cronin: no updates, sadly. But if you're interested in making the PR, I can give you pointers! I believe it's fairly easy to add the feature.

**Comment by robert-cronin** [comment]
@ephread I'd love to, I might use what @kientux has provided as inspiration for a new method of FlowManager if thats okay with everyone :)
I'll try and find time to submit PR this week if time and complexity permits!

**Comment by ephread** [comment]
Closing, since it was implemented by #182.
